index.d.ts

```diff
@@ -900,6 +932,34 @@ declare namespace Eris {
     systemChannelID?: string;
     verificationLevel?: VerificationLevel;
   }
+  interface GuildScheduledEventMetadata {
+    location?: string;
+  }
+  interface GuildScheduledEventEditOptions<T extends GuildScheduledEventEntityTypes = GuildScheduledEventEntityTypes> {
+    channelID: T extends Constants["GuildScheduledEventEntityTypes"]["EXTERNAL"] ? null | undefined : string | undefined;
+    description?: string | null;
+    entityMetadata: T extends Constants["GuildScheduledEventEntityTypes"]["EXTERNAL"] ? Required<GuildScheduledEventMetadata> : GuildScheduledEventMetadata | null | undefined;
+    entityType?: T;
+    image?: string;
+    name?: string;
+    privacyLevel?: GuildScheduledEventPrivacyLevel;
+    scheduledEndTime: T extends Constants["GuildScheduledEventEntityTypes"]["EXTERNAL"] ? Date : Date | undefined
+    scheduledStartTime?: Date;
+    status?: GuildScheduledEventStatus;
+  }
+  interface GuildScheduledEventOptions<T extends GuildScheduledEventEntityTypes = GuildScheduledEventEntityTypes> extends Omit<GuildScheduledEventEditOptions<T>, "status"> {
+    channelID: T extends Constants["GuildScheduledEventEntityTypes"]["EXTERNAL"] ? undefined : string;
+    entityMetadata: T extends Constants["GuildScheduledEventEntityTypes"]["EXTERNAL"] ? Required<GuildScheduledEventMetadata> : GuildScheduledEventMetadata | undefined;
+    entityType: T;
+    name: string;
+    privacyLevel: GuildScheduledEventPrivacyLevel;
+    scheduledStartTime: Date;
+  }
```

Comment:
When you have `EXTERNAL` as EntityType for example, you still have to define things like `channelID`. Only way to fix that is by creating separate interfaces for `EXTERNAL` and the other EntityTypes, this way you can make separate optional properties. 

Not quite sure if that's worth it, but making the properties fully optional ain't ideal either. Perhaps @bsian03 can share his opinion about this.

Reply:
Discord says that `channel_id` is optional when the entity type is `EXTERNAL`. I could probably type it as `string | undefined` though, but that would also need some not-so-pretty interface modifying

Reply:
The issue is though. If you don't use `property?: type`, it will still require you to specify it in the object, but as undefined or null (whatever the type is).

Reply:
_unless you use `never` ofcourse..._